File: Tl2.4 RESTFULL_DOTNET_EUREKA_BANK_GR08/02.CLICON/CLICON_EUREKA_GR08/ec.edu.monster.view/MainView.cs
using System;
using System.Text;
using CLICON_EUREKA_GR08.ec.edu.monster.controller;

namespace CLICON_EUREKA_GR08.ec.edu.monster.view
{
    public class MainView
    {
        private static MainController mainController = new MainController();

        public static void Iniciar()
        {
            while (true)
            {
                if (mainController.GetUsuarioActual() == null)
                {
                    MenuLogin();
                }
                else
                {
                    MenuPrincipal();
                }
            }
        }

        private static void MenuLogin()
        {
            StringBuilder contenido = new StringBuilder();
            contenido.AppendLine("1. Iniciar Sesión");
            contenido.AppendLine("2. Salir");
            Console.WriteLine(EncloseInBox("EUREKABANK - RESTFULL", contenido.ToString()));
            Console.Write("Seleccione una opción: ");

            string opcionStr = Console.ReadLine();
            if (int.TryParse(opcionStr, out int opcion))
            {
                switch (opcion)
                {
                    case 1:
                        IniciarSesion();
                        break;
                    case 2:
                        Console.WriteLine("Gracias por usar nuestros servicios. ¡Hasta luego!");
                        Environment.Exit(0);
                        break;
                    default:
                        Console.WriteLine("Opción inválida. Intente nuevamente.");
                        break;
                }
            }
            else
            {
                Console.WriteLine("Opción inválida. Intente nuevamente.");
            }
        }

        private static void IniciarSesion()
        {
            Console.Write("Ingrese su nombre de usuario: ");
            string username = Console.ReadLine();
            Console.Write("Ingrese su contraseña: ");
            string password = Console.ReadLine();

            if (mainController.IniciarSesion(username, password))
            {
                Console.WriteLine("Inicio de sesión exitoso.");
            }
            else
            {
                Console.WriteLine("Credenciales incorrectas. Intente nuevamente.");
            }
        }

        private static void MenuPrincipal()
        {
            StringBuilder contenido = new StringBuilder();
            contenido.AppendLine("1. Realizar Depósito");
            contenido.AppendLine("2. Realizar Retiro");
            contenido.AppendLine("3. Realizar Transferencia");
            contenido.AppendLine("4. Ver Movimientos");
            contenido.AppendLine("5. Ver Datos Cuenta");
            contenido.AppendLine("6. Cerrar Sesión");
            Console.WriteLine(EncloseInBox("MENÚ PRINCIPAL", contenido.ToString()));
            Console.Write("Seleccione una opción: ");

            string opcionStr = Console.ReadLine();
            if (int.TryParse(opcionStr, out int opcion))
            {
                switch (opcion)
                {
                    case 1:
                        RealizarDeposito();
                        break;
                    case 2:
                        RealizarRetiro();
                        break;
                    case 3:
                        RealizarTransferencia();
                        break;
                    case 4:
                        VerMovimientos();
                        break;
                    case 5:
                        VerDatosCuenta();
                        break;
                    case 6:
                        CerrarSesion();
                        break;
                    default:
                        Console.WriteLine("Opción inválida. Intente nuevamente.");
                        break;
                }
            }
            else
            {
                Console.WriteLine("Opción inválida. Intente nuevamente.");
            }
        }

        private static void RealizarDeposito()
        {
            Console.Write("Ingrese el número de cuenta: ");
            string cuenta = Console.ReadLine();
            Console.Write("Ingrese el monto a depositar: ");
            string monto = Console.ReadLine();

            if (mainController.RealizarDeposito(cuenta, monto, "DEP", null))
            {
                Console.WriteLine("Depósito realizado con éxito.");
            }
            else
            {
                Console.WriteLine("Error al realizar el depósito.");
            }
        }

        private static void RealizarRetiro()
        {
            Console.Write("Ingrese el número de cuenta: ");
            string cuenta = Console.ReadLine();
            Console.Write("Ingrese el monto a retirar: ");
            string monto = Console.ReadLine();

            if (mainController.RealizarRetiro(cuenta, monto))
            {
                Console.WriteLine("Retiro realizado con éxito.");
            }
            else
            {
                Console.WriteLine("Error al realizar el retiro.");
            }
        }

        private static void RealizarTransferencia()
        {
            Console.Write("Ingrese el número de cuenta origen: ");
            string cuentaOrigen = Console.ReadLine();
            Console.Write("Ingrese el número de cuenta destino: ");
            string cuentaDestino = Console.ReadLine();
            Console.Write("Ingrese el monto a transferir: ");
            string monto = Console.ReadLine();

            if (mainController.RealizarTransferencia(cuentaOrigen, cuentaDestino, monto))
            {
                Console.WriteLine("Transferencia realizada con éxito.");
            }
            else
            {
                Console.WriteLine("Error al realizar la transferencia.");
            }
        }

        private static void VerMovimientos()
        {
            Console.Write("Ingrese el número de cuenta: ");
            string cuenta = Console.ReadLine();

            string resultado = mainController.VerMovimientos(cuenta);
            Console.WriteLine(resultado);
        }

        private static void VerDatosCuenta()
        {
            Console.Write("Ingrese el número de cuenta: ");
            string cuenta = Console.ReadLine();

            string resultado = mainController.VerDatosCuenta(cuenta);
            Console.WriteLine(resultado);
        }

        private static void CerrarSesion()
        {
            mainController.CerrarSesion();
            Console.WriteLine("Sesión cerrada exitosamente.");
        }

        // Utilidades de recuadro ASCII (coherentes con el estilo del controlador)
        private static string EncloseInBox(string title, string multilineContent)
        {
            string[] lines = multilineContent.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
            int maxLineLength = title.Length;
            foreach (string line in lines)
            {
                if (line.Length > maxLineLength)
                {
                    maxLineLength = line.Length;
                }
            }
            int innerWidth = Math.Max(maxLineLength, title.Length);
            string horizontal = Repeat('-', innerWidth + 2);

            StringBuilder box = new StringBuilder();
            box.Append('+').Append(horizontal).Append("+").AppendLine();
            box.Append('|').Append(PadCenter(" " + title + " ", innerWidth + 2)).Append('|').AppendLine();
            box.Append('+').Append(horizontal).Append("+").AppendLine();
            foreach (string line in lines)
            {
                box.Append('|').Append(PadRight(" " + line, innerWidth + 2)).Append('|').AppendLine();
            }
            box.Append('+').Append(horizontal).Append('+');
            return box.ToString();
        }

        private static string Repeat(char ch, int count)
        {
            return new string(ch, count);
        }

        private static string PadRight(string text, int width)
        {
            if (text.Length >= width)
            {
                return text;
            }
            return text.PadRight(width);
        }

        private static string PadCenter(string text, int width)
        {
            if (text.Length >= width)
            {
                return text;
            }
            int totalPadding = width - text.Length;
            int left = totalPadding / 2;
            int right = totalPadding - left;
            return new string(' ', left) + text + new string(' ', right);
        }
    }
}

File: TI2.1  SOAP_JAVA_EUREKA_BANK_GR08/02.CLICON/CLICON_EUREKA_GR08/src/main/java/ec/edu/monster/vista/MainView.java
package ec.edu.monster.vista;

import ec.edu.monster.controlador.MainController;

import java.util.Scanner;

public class MainView {
    private static Scanner scanner = new Scanner(System.in);
    private static MainController mainController = new MainController();

    public static void main(String[] args) {
        while (true) {
            if (mainController.getUsuarioActual() == null) {
                menuLogin();
            } else {
                menuPrincipal();
            }
        }
    }

    private static void menuLogin() {
        StringBuilder contenido = new StringBuilder();
        contenido.append("1. Iniciar Sesión\n");
        contenido.append("2. Salir\n");
        System.out.println(encloseInBox("EUREKABANK - SOAP", contenido.toString()));
        System.out.print("Seleccione una opción: ");

        int opcion = scanner.nextInt();
        scanner.nextLine(); // Limpiar buffer

        switch (opcion) {
            case 1 -> iniciarSesion();
            case 2 -> {
                System.out.println("Gracias por usar nuestros servicios. ¡Hasta luego!");
                System.exit(0);
            }
            default -> System.out.println("Opción inválida. Intente nuevamente.");
        }
    }

    private static void iniciarSesion() {
        System.out.print("Ingrese su nombre de usuario: ");
        String username = scanner.nextLine();
        System.out.print("Ingrese su contraseña: ");
        String password = scanner.nextLine();

        if (mainController.iniciarSesion(username, password)) {
            System.out.println("Inicio de sesión exitoso.");
        } else {
            System.out.println("Credenciales incorrectas. Intente nuevamente.");
        }
    }

    private static void menuPrincipal() {
        StringBuilder contenido = new StringBuilder();
        contenido.append("1. Realizar Depósito\n");
        contenido.append("2. Realizar Retiro\n");
        contenido.append("3. Realizar Transferencia\n");
        contenido.append("4. Ver Movimientos\n");
        contenido.append("5. Ver Datos Cuenta\n");
        contenido.append("6. Cerrar Sesión\n");
        System.out.println(encloseInBox("MENÚ PRINCIPAL", contenido.toString()));
        System.out.print("Seleccione una opción: ");

        int opcion = scanner.nextInt();
        scanner.nextLine(); // Limpiar buffer

        switch (opcion) {
            case 1 -> realizarDeposito();
            case 2 -> realizarRetiro();
            case 3 -> realizarTransferencia();
            case 4 -> verMovimientos();
            case 5 -> verDatosCuenta();
            case 6 -> cerrarSesion();
            default -> System.out.println("Opción inválida. Intente nuevamente.");
        }
    }

private static void realizarDeposito() {
    System.out.print("Ingrese el número de cuenta: ");
    String cuenta = scanner.nextLine();
    System.out.print("Ingrese el monto a depositar: ");
    String monto = scanner.nextLine();

    if (mainController.realizarDeposito(cuenta, monto)) {
        System.out.println("Depósito realizado con éxito.");
    } else {
        System.out.println("Error al realizar el depósito.");
    }
}



    private static void realizarRetiro() {
        System.out.print("Ingrese el número de cuenta: ");
        String cuenta = scanner.nextLine();
        System.out.print("Ingrese el monto a retirar: ");
        String monto = scanner.nextLine();

        if (mainController.realizarRetiro(cuenta, monto)) {
            System.out.println("Retiro realizado con éxito.");
        } else {
            System.out.println("Error al realizar el retiro.");
        }
    }


private static void realizarTransferencia() {
    System.out.print("Ingrese el número de cuenta origen: ");
    String cuentaOrigen = scanner.nextLine();
    System.out.print("Ingrese el número de cuenta destino: ");
    String cuentaDestino = scanner.nextLine();
    System.out.print("Ingrese el monto a transferir: ");
    String monto = scanner.nextLine();

    if (mainController.realizarTransferencia(cuentaOrigen, cuentaDestino, monto)) {
        System.out.println("Transferencia realizada con éxito.");
    } else {
        System.out.println("Error al realizar la transferencia.");
    }
}

    private static void verMovimientos() {
        System.out.print("Ingrese el número de cuenta: ");
        String cuenta = scanner.nextLine();

        String resultado = mainController.verMovimientos(cuenta);
        System.out.println(resultado);
    }

    private static void verDatosCuenta() {
        System.out.print("Ingrese el número de cuenta: ");
        String cuenta = scanner.nextLine();

        String resultado = mainController.verDatosCuenta(cuenta);
        System.out.println(resultado);
    }

    private static void cerrarSesion() {
        mainController.cerrarSesion();
        System.out.println("Sesión cerrada exitosamente.");
    }

    // Utilidades de recuadro ASCII (coherentes con el estilo del controlador)
    private static String encloseInBox(String title, String multilineContent) {
        String[] lines = multilineContent.split("\\r?\\n");
        int maxLineLength = title.length();
        for (String line : lines) {
            if (line.length() > maxLineLength) {
                maxLineLength = line.length();
            }
        }
        int innerWidth = Math.max(maxLineLength, title.length());
        String horizontal = repeat('-', innerWidth + 2);

        StringBuilder box = new StringBuilder();
        box.append('+').append(horizontal).append("+\n");
        box.append('|').append(padCenter(" " + title + " ", innerWidth + 2)).append('|').append("\n");
        box.append('+').append(horizontal).append("+\n");
        for (String line : lines) {
            box.append('|').append(padRight(" " + line, innerWidth + 2)).append('|').append("\n");
        }
        box.append('+').append(horizontal).append('+');
        return box.toString();
    }

    private static String repeat(char ch, int count) {
        StringBuilder sb = new StringBuilder(count);
        for (int i = 0; i < count; i++) {
            sb.append(ch);
        }
        return sb.toString();
    }

    private static String padRight(String text, int width) {
        if (text.length() >= width) {
            return text;
        }
        StringBuilder sb = new StringBuilder(width);
        sb.append(text);
        while (sb.length() < width) {
            sb.append(' ');
        }
        return sb.toString();
    }

    private static String padCenter(String text, int width) {
        if (text.length() >= width) {
            return text;
        }
        int totalPadding = width - text.length();
        int left = totalPadding / 2;
        int right = totalPadding - left;
        StringBuilder sb = new StringBuilder(width);
        for (int i = 0; i < left; i++) sb.append(' ');
        sb.append(text);
        for (int i = 0; i < right; i++) sb.append(' ');
        return sb.toString();
    }
}